### ISSUE TITLE
Loss function is giving different results from tensorflow implementation

### ISSUE THREAD

**Issue by samedii** [Spijkervet/SimCLR#18]
I noticed that your loss function is behaving differently from the original tensorflow implementation. It also seems like every pytorch implementation may have copied from each other because they are all giving the same result

Here is a reimplementation of the tensorflow version if you want to try it
```
import torch
import torch.nn.functional as F


def nt_cross_entropy(x, y, temperature=0.5):
    '''
    normalized temperature-scaled cross entropy loss
    
    tensorflow: https://github.com/google-research/simclr/blob/master/objective.py
    '''
    normalized_x = F.normalize(x, dim=1)
    normalized_y = F.normalize(y, dim=1)

    auto_x = normalized_x @ normalized_x.t() / temperature
    auto_y = normalized_y @ normalized_y.t() / temperature

    corr_xy = normalized_x @ normalized_y.t() / temperature
    corr_yx = normalized_y @ normalized_x.t() / temperature

    targets = torch.arange(auto_x.shape[0]).to(auto_x.device)
    big_diagonal = torch.eye(auto_x.shape[0], device=auto_x.device) * 1e9
    return (
        F.cross_entropy(
            torch.cat([corr_xy, auto_x - big_diagonal], dim=1),
            targets,
        )
        + F.cross_entropy(
            torch.cat([corr_yx, auto_y - big_diagonal], dim=1),
            targets,
        )
    )


def test_nt_cross_entropy():
    x = torch.tensor([[1.0, 2.0], [3.0, -2.0], [1.0, 5.0]])
    y = torch.tensor([[1.0, 0.75], [2.8, -1.75], [1.0, 4.7]])
    assert abs(nt_cross_entropy(x, y, temperature=1.0).item() - 2.2654006) <= 1e-5

    z = torch.tensor([[1.0, 1.75], [2.8, -1.75], [1.0, 4.7]])
    assert abs(nt_cross_entropy(x, z, temperature=1.0).item() - 2.1991892) <= 1e-5
```

**Comment by gauenk** [comment]
I did some inspection with the current code. I think the pytorch code in this repo ends up being the same as yours, but divides your code's error by 2. It is a bit complicated to follow, but I worked out the logic. I feel the code in this github is proportionally correct, missing only a divide by 2 constant.

**Comment by samedii** [comment]
Nice! :) Thanks for checking, I'll close this then
